### PR TITLE
refactor(web): asset viewer actions

### PR DESCRIPTION
**Pull request by michelheusschen**
Create individual components for actions. These components make their own API calls and register their own shortcuts. This declutters the main `AssetViewer` component and makes it easier to disable specific actions. Also fixes the issue that you can still perform actions like favorite, delete, download, etc. while the slideshow is active by using shortcuts.

### CODE REVIEW

**Review by danieldietzler — APPROVED** [review]
The code looks really good to me!

File: e2e/src/web/specs/asset-viewer/slideshow.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    await utils.resetDatabase();
+    admin = await utils.adminSetup();
+    asset = await utils.createAsset(admin.accessToken);
+  });
+
+  const openSlideshow = async (page: Page) => {
+    await page.goto(`/photos/${asset.id}`);
+    await page.waitForSelector('#immich-asset-viewer');
+    await page.getByRole('button', { name: 'More' }).click();
+    await page.getByRole('menuitem', { name: 'Slideshow' }).click();
```
[the remark this answers is not in the document]
Comment (michelheusschen) [reply e2e/src/web/specs/asset-viewer/slideshow.e2e-spec.ts:20]: Yeah using the `dev` language would definitely be better, but I'm not sure how to properly do that using playwright. Maybe I can look at that in another PR.

**Review by jrasm91 — APPROVED** [review]
This is great. What do you think about eventually using a slot for the actions instead of a million show* boolean props?

File: e2e/src/web/specs/asset-viewer/slideshow.e2e-spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (danieldietzler) [reply e2e/src/web/specs/asset-viewer/slideshow.e2e-spec.ts:20]: I guess worst case you could actually navigate to that page and switch it before all other tests run. But yeah, hopefully there's some better way to do that. Would appreciate it! :)

[the remark this answers is not in the document]
Comment (jrasm91) [reply e2e/src/web/specs/asset-viewer/slideshow.e2e-spec.ts:20]: I think you can make a script/step with to local storage.

**Comment by michelheusschen** [comment]
> This is great. What do you think about eventually using a slot for the actions instead of a million show* boolean props?

You can't pass slots conditionally without using `svelte:fragment`, which makes that a bit ugly. I did create a slot for motion photos, but I'm not sure it makes much sense for other actions. I've also moved more logic to the navbar to reduce the need for some of these props.

**Comment by jrasm91** [comment]
I thought we could pass all the items in a single slot. I don't think internally it needs to know about every individual menu item

**Pull request merged**